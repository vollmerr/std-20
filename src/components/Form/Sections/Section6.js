import React from 'react';
import T from 'prop-types';
import { FieldDate, FieldInput } from 'state-template';

import schema from '../schema';

const Section6 = ({ disabled }) => (
  <div className={'row'} role={'group'} aria-labelledby={'section6__header'}>
    <div id={'section6__header'} className={'col-md-12'}>6. CATR/ATR Information</div>

    <FieldInput {...schema.catrName} className={'col-md-4'} disabled={disabled} />
    <FieldInput {...schema.catrEmail} className={'col-md-4'} disabled={disabled} />
    <FieldInput {...schema.catrPhone} className={'col-md-2'} disabled={disabled} />
    <FieldInput {...schema.catrFax} className={'col-md-2'} disabled={disabled} />

    <FieldInput {...schema.catrAddress} className={'col-md-4'} disabled={disabled} />
    <FieldInput {...schema.catrCity} className={'col-md-4'} disabled={disabled} />
    <FieldInput {...schema.catrState} className={'col-md-2'} disabled={disabled} />
    <FieldInput {...schema.catrZip} className={'col-md-2'} disabled={disabled} />

    <FieldInput {...schema.catrSignature} className={'col-md-4'} disabled={disabled} />
    <FieldInput {...schema.catrTitle} className={'col-md-4'} disabled={disabled} />
    <FieldDate {...schema.catrDate} className={'col-md-4'} disabled={disabled} />
  </div>
);

Section6.propTypes = {
  disabled: T.bool.isRequired,
};

export default Section6;
