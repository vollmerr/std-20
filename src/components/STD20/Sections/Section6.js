import React from 'react';
import { FieldDate, FieldInput } from 'state-template';

import schema from '../schema';

const Section6 = () => (
  <fieldset className={'form-group'}>
    <div className={'row'}>
      <legend className={'col-md-12'}>6. CATR/ATR Information</legend>

      <FieldInput {...schema.catrName} className={'col-md-4'} />
      <FieldInput {...schema.catrEmail} className={'col-md-4'} />
      <FieldInput {...schema.catrPhone} className={'col-md-2'} />
      <FieldInput {...schema.catrFax} className={'col-md-2'} />

      <FieldInput {...schema.catrAddress} className={'col-md-4'} />
      <FieldInput {...schema.catrCity} className={'col-md-4'} />
      <FieldInput {...schema.catrState} className={'col-md-2'} />
      <FieldInput {...schema.catrZip} className={'col-md-2'} />

      <FieldInput {...schema.catrSignature} className={'col-md-4'} />
      <FieldInput {...schema.catrTitle} className={'col-md-4'} />
      <FieldDate {...schema.catrDate} className={'col-md-4'} />
    </div>
  </fieldset>
);

export default Section6;
