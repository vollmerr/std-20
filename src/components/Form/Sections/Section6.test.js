import React from 'react';
import { shallow } from 'enzyme';

import Section6 from './Section6';

const props = {
  disabled: false,
};

let wrapper;
describe('Section6', () => {
  beforeEach(() => {
    wrapper = shallow(<Section6 {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set all fields as disabled', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.find({ disabled: true }).length).toBe(11);
  });
});
