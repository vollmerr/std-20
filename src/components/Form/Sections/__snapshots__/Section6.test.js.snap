// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section6 should render correctly 1`] = `
<div
  aria-labelledby="section6__header"
  className="row"
  role="group"
>
  <div
    className="col-md-12"
    id="section6__header"
  >
    6. CATR/ATR Information
  </div>
  <Component
    className="col-md-4"
    disabled={false}
    label="Name"
    name="catrName"
  />
  <Component
    className="col-md-4"
    disabled={false}
    label="Email Address"
    name="catrEmail"
  />
  <Component
    className="col-md-2"
    disabled={false}
    label="Phone Number"
    name="catrPhone"
    normalize={[Function]}
  />
  <Component
    className="col-md-2"
    disabled={false}
    label="Fax Number"
    name="catrFax"
    normalize={[Function]}
  />
  <Component
    className="col-md-4"
    disabled={false}
    label="Address"
    name="catrAddress"
  />
  <Component
    className="col-md-4"
    disabled={false}
    label="City"
    name="catrCity"
  />
  <Component
    className="col-md-2"
    disabled={false}
    label="State"
    name="catrState"
  />
  <Component
    className="col-md-2"
    disabled={false}
    label="Zip Code"
    name="catrZip"
    normalize={[Function]}
  />
  <Component
    className="col-md-4"
    disabled={true}
    helpText="This request complies with State telecommunication policies."
    label="Signature"
    name="signature"
  />
  <Component
    className="col-md-4"
    disabled={false}
    label="Title"
    name="catrTitle"
  />
  <Component
    className="col-md-4"
    disabled={false}
    label="Date"
    name="catrDate"
  />
</div>
`;
